### PR TITLE
feat: implement quick filters for the new logs explorer page 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

- add quick filters to the logs explorer page
- detailed information can be found here - https://www.notion.so/signoz/logs-quick-filters-c39ece1c07524e72b6163d7678ed16a4 

#### Related Issues / PR's

fixes https://github.com/SigNoz/signoz/issues/3536 

#### Screenshots

https://github.com/user-attachments/assets/7a72b994-8837-43cb-ace8-fc0063e31ca3

https://github.com/user-attachments/assets/1abe094b-b294-4d91-a9d9-86f3847ca61f

#### Affected Areas and Manually Tested Areas

- different cases of logs explorer filters with sync and non synced versions
- none of the changes translates to traces page and dashboards.

### CODE REVIEW

**Comment by ankitnayan** [comment]
Please include `deployment.environment` in the fast filters as `Environment`

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
* Selecting all back again doesn't (by manually checking boxing) not necessarily clears the filter always ( video attached)
* Adding a filter like '=' manually and then trying to add from quick filter doesn't work always(video attached)
* I am personally not convinced with how the quick filters work for multiple queries, either it should be all or a better way to let the user know that it is applied for this specific query.  ( also where to click for that query to be highlited is a concern again)

--- design improvements
* better way to let the user know that some filters in this query are added from quick filters ( eg some colour or something else)



https://github.com/user-attachments/assets/79eab1c8-f2ea-4624-92f7-6079f1412af9


https://github.com/user-attachments/assets/6067abaa-2b79-4b6f-bd27-d5e32cf16c69

**Comment by vikrantgupta25** [comment]
@GeekBoySupreme  :- need your inputs for the design implications for the sync action

**Comment by vikrantgupta25** [comment]
@nityanandagohain 

> Selecting all back again doesn't (by manually checking boxing) not necessarily clears the filter always ( video attached)/'

expected behaviour as discussed on call. for the `= and !=` operator i ll make the required changes

**Comment by vikrantgupta25** [comment]
@nityanandagohain made the changes for the `=` and `!=` operator as well.

**Comment by GeekBoySupreme** [comment]
@vikrantgupta25 not a lot of change from how you were showing the active query for applying fast filters -

image

The color of the query name on the fast filters sidebar should be the same as the color on the query stack and also on the right minimap.

image

Clicking on `A` or whichever is the active query name on the fast filters pane, you scroll to bring that specific query into focus from the query stack.

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
The '=', '!=' changes looks good but it has created a new bug, not sure if it's due to this,


The logs page now doesn't render correctly

image

a lot of space is left out below.

**Comment by vikrantgupta25** [comment]
@nityanandagohain @GeekBoySupreme addressed the issues and the design changes

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
The is some css edge case, please add a fix for it

https://github.com/user-attachments/assets/73f5684c-a99c-45c4-8afc-cfe1247574de

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
Should have tested this earlier, but selecting an existing view and then apply filters the frontend crashes.

image

**Review by nityanandagohain — APPROVED** [review]

**Review by nityanandagohain — APPROVED** [review]

File: frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.styles.scss
Comment (SagarRajput-7) [inline frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.styles.scss:1]: remove empty file

Comment (SagarRajput-7) [reply frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.styles.scss:1]: ohh, this is for future use - for slider got it

File: frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { IQuickFiltersConfig } from 'components/QuickFilters/QuickFilters';
+
+interface ISliderProps {
+	filter: IQuickFiltersConfig;
+}
+
+// not needed for now build when required
+export default function Slider(props: ISliderProps): JSX.Element {
+	const { filter } = props;
+	console.log(filter);
```
Comment (rahulkeswani101) [inline frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.tsx:12]: Can you remove this console statement? 
Rest LGTM.

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.tsx (hunk 1 above)
Comment (vikrantgupta25) [reply frontend/src/components/QuickFilters/FilterRenderers/Slider/Slider.tsx:12]: this is no where used , will start giving eslint errors for unused filters. will remove when we consume this.

**Review by rahulkeswani101 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**